pkgs/development/libraries/mesa/default.nix

```diff
@@ -233,20 +238,20 @@ self = stdenv.mkDerivation {
     python3Packages.python # for shebang
   ] ++ lib.optionals haveWayland [ wayland wayland-protocols ]
     ++ lib.optionals stdenv.isLinux [ libomxil-bellagio libva-minimal udev lm_sensors ]
-    ++ lib.optionals enableOpenCL [ llvmPackages.libclc llvmPackages.clang llvmPackages.clang-unwrapped spirv-llvm-translator ]
+    ++ lib.optionals needsRust [ llvmPackages.libclc llvmPackages.clang llvmPackages.clang-unwrapped spirv-llvm-translator ]
```

Comment:
Wrong conditional?

Reply:
Isn't `rusticl` enabled with `enableOpenCL`?

Reply:
It's not the only thing enabled with enableOpenCL. There's also the older Clover frontend.

Reply:
NVK and Rusticl needs Rustc and `-Drust_std=2021`
NVK and Clover need libclc
:eyes:

Reply:
Then maybe we should just give up on making this stuff conditional?

Reply:
We will want these deps for the 32-bits NVK, but (at least right now) we don't build Clover or Rusticl in our 32-bits mesa.

Reply:
Maybe we should do that too.

Reply:
So options are:
1. Weird conditional that mix Rust and LLVM;
2. Always pull all deps (now including Rust, Clang, LibCLC, and SPIR-V) but keep Clover and Rusticl meson-options switchable;
3. Pull all deps and build the same stuff in all architectures.

Reply:
I'm leaning towards 3, if people really want to turn it off, they can always override mesonFlags.